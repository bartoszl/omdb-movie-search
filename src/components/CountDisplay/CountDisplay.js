import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const CountText = styled.p`
  color: white;
  font-size: 1.25rem;
  text-align: center;
  margin-bottom: 2rem;
  margin-top: 0;
`;

const CountDisplay = ({ count }) => (
  <CountText>
    { `Found ${count} movies.`}
  </CountText>
);

CountDisplay.propTypes = {
  count: PropTypes.number,
};

CountDisplay.defaultProps = {
  count: 0,
};

export default CountDisplay;
