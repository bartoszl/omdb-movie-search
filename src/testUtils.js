import React from 'react';
import PropTypes from 'prop-types';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import theme from './config/theme';

const TestProviders = ({ children }) => (
  <ThemeProvider theme={theme}>
    <MemoryRouter>
      {children}
    </MemoryRouter>
  </ThemeProvider>
);

TestProviders.propTypes = {
  children: PropTypes.node.isRequired,
};

const customRender = (ui, options) => render(ui, { wrapper: TestProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
