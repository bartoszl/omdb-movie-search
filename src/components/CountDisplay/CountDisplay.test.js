import React from 'react';
import { render, cleanup, screen } from '../../testUtils';
import CountDisplay from './CountDisplay';

describe('CountDisplay', () => {
  const count = 1;

  afterEach(() => {
    cleanup();
  });

  test('dislpays 0 as default count', () => {
    render((
      <CountDisplay count={count} />
    ));

    const text = screen.findByText('Found 0 movies.');

    expect(text).toBeTruthy();
  });

  test('dislpays correct count', () => {
    render((
      <CountDisplay count={count} />
    ));

    const text = screen.findByText(`Found ${count} movies.`);

    expect(text).toBeTruthy();
  });

  test('matches snapshot', () => {
    const { asFragment } = render((
      <CountDisplay count={count} />
    ));

    expect(asFragment()).toMatchSnapshot();
  });
});
