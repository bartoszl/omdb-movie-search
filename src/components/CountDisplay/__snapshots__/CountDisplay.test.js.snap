// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountDisplay matches snapshot 1`] = `
<DocumentFragment>
  <p
    class="sc-bdnylx kKvyBw"
  >
    Found 1 movies.
  </p>
</DocumentFragment>
`;
